examples/gno.land/r/gov/dao/voter.gno

```diff
@@ -0,0 +1,57 @@
+package govdao
+
+import (
+	"std"
+)
+
+const YAY = "YES"
+const NAY = "NO"
+
+func NewPercentageVoter(percent int) *PercentageVoter {
+	if percent < 0 || percent > 100 {
+		panic("percent value must be between 0 and 100")
+	}
+
+	return &PercentageVoter{
+		percentage: percent,
+	}
+}
+
+// PercentageVoter is a system based on the amount of received votes.
+// When the specified treshold is reached, the voting process finishes.
+type PercentageVoter struct {
+	percentage int
+
+	yes int
+	no  int
+}
+
+func (pv *PercentageVoter) Accepted(voters []std.Address) bool {
+	if len(voters) == 0 {
+		return true // special case
+	}
+
+	return (pv.yes/len(voters))*100 >= pv.percentage
+}
+
+func (pv *PercentageVoter) Finished(voters []std.Address) bool {
+	return pv.yes+pv.no >= len(voters)
+}
+
+func (pv *PercentageVoter) Vote(voters []std.Address, flag string) {
```

Comment:
How do you guard against double voting?

Reply:
Great catch, not guarded. I'll add a test for that case.

Reply:
done.